In [1]:
import sqlite3

import pandas as pd

#Connection à la base de donnée
connection = sqlite3.connect('babies.db')

#connection à la base de donnée
cursor = connection.cursor()

#Création et remplissage des tables de notre base de données 
Les requêtes de la création manuelle sont disponibles dans le rapport associé.

Etant donné que la base de donnée est open-data et que n'anonymat des individus est assuré nous n'avons pas pu avoir les identifiants réels des individus donc nous avons utilisé les numéros de ligne à la place pour garder des résultats cohérents avec la partie python

# Importation des données pour remplir les tables

In [11]:
bebe =pd.read_csv("bebe.csv")
mere = pd.read_csv("mere.csv")
pere = pd.read_csv("pere.csv")
zone = pd.read_csv("zone.csv")

bebe.to_sql("bebe",connection,if_exists="replace")
mere.to_sql("mere",connection,if_exists="replace")
pere.to_sql("pere",connection,if_exists="replace")
zone.to_sql("zone",connection,if_exists="replace")

/usr/local/lib/python3.6/dist-packages/pandas/core/generic.py:2615: UserWarning: The spaces in these column names will not be changed. In pandas versions < 0.14, spaces were converted to underscores.
  method=method,


#Les requêtes de filtrage

In [4]:
#Tous le bébés avec un poids plus petit que 2.5 kg
req ="""
SELECT * 
FROM bebe
WHERE poids_bebe <2.5
"""
pd.read_sql_query(req,connection)

index  id_bebe  ordre_naissance  ...  mere_race  id_mere  id_pere
0             3        4                6  ...         10        4        4
1            20       22                1  ...         10       22       22
2            62       65                3  ...         10       65       65
3            70       75                3  ...         61       75       75
4            80       85                3  ...         40       85       85
...         ...      ...              ...  ...        ...      ...      ...
155129  2053397  2425135                4  ...         10  2425135  2425135
155130  2053429  2425172                2  ...         10  2425172  2425172
155131  2053431  2425174                1  ...         10  2425174  2425174
155132  2053437  2425180                1  ...         10  2425180  2425180
155133  2053450  2425195                2  ...         61  2425195  2425195

[155134 rows x 15 columns]

In [5]:
#Nombre de jumeaux dont la mère n'a pas suivi de consultation prénatale pendant leur grossesse et leur poids moyen
req = """
SELECT COUNT(*) as nombre_sans_visite , AVG(poids_bebe) as poids_moyen
FROM bebe
WHERE previs =2 AND dplural = 2 
"""
pd.read_sql_query(req,connection)

nombre_sans_visite  poids_moyen
0                 455     1.870523

il ya 455 enfants dont la mère n'a pas suivi de consultation prénatale pendant toute la durée de leur grossesse

# Requete utilisant group by

In [7]:
#Liste poids moyen des bébé par sexe
req ="""
SELECT sex, AVG(poids_bebe) as poids_moyen
FROM bebe
GROUP BY sex
"""
pd.read_sql_query(req,connection)

sex  poids_moyen
0  0.0     3.225643
1  1.0     3.340414

Le poids moyen des enfants de sexe masculin est 3,3 kg et celui des enfants de sexe masculin est 3,3 Kg

#Requêtes avec jointures

In [8]:
#Liste bébés dont les mères sont obèses
#Une personne obèse a un imc de plus de 35
req = """
SELECT * 
FROM bebe
INNER JOIN mere ON bebe.id_mere =  mere.id_mere
WHERE imc >= 35 
"""
pd.read_sql_query(req,connection)

index  id_bebe  ...  mere_fumeuse  mere_race
0             3        4  ...           0.0         10
1            24       26  ...           1.0         10
2            45       47  ...           0.0         61
3            50       53  ...           0.0         61
4            53       56  ...           0.0         10
...         ...      ...  ...           ...        ...
256971  2053494  2425244  ...           0.0         20
256972  2053500  2425252  ...           0.0         10
256973  2053504  2425256  ...           0.0         10
256974  2053511  2425264  ...           1.0         10
256975  2053512  2425267  ...           0.0         10

[256976 rows x 23 columns]

#Requête avec HAVING

In [26]:
#Liste des zones ayant compté au moins 15000 enfants au cours de l'année
req ="""
SELECT code_zone,label , COUNT(*) as nombre_bebe
FROM bebe
INNER JOIN mere ON bebe.id_mere = mere.id_mere
INNER JOIN zone ON mere.code_zone = zone.code_zone
GROUP BY zone.code_zone
HAVING nombre_bebe >=15000
ORDER BY age_mere desc 
"""
pd.read_sql_query(req,connection)

code_zone                   label  nombre_bebe
0         61                 Inconnu       107776
1         40  Asiatique (uniquement)       252810
2         20       Noir (uniquement)       549514
3         41      NHOPI (uniquement)        56480
4         10      Blanc (uniquement)      3093332
5         30       AIAN (uniquement)        34640